runner_test.go

```diff
@@ -188,4 +200,58 @@ func TestRunByTestID(t *testing.T) {
 		createRun()
 		ts.Close()
 	}
+	runTestInBackground = tempTestInBackground
+}
+func TestRunInForeground(t *testing.T) {
+	var percent int
+	tempTestInBackground := runTestInBackground
+	runTestInBackground = false
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```

Comment:
This testing method looks good overall, but there are a couple minor issues:

1. It's very long (to the point where it's hard to read all at once). Try moving some stuff into helper functions.
2. I haven't run it, but it seems like the delay might happen in the test. It's never a good idea to have `sleep`s in your tests since it slows them down unnecessarily; you might want to move the polling interval into a global variable or something.
3. You're only testing the "happy path", and even then you have very few assertions (for quite a lot of work). There are at least 3 cases that are important:

 a. "Happy case" (test runs and passes), which you test pretty well.
 b. Test fails and fail-fast == false (which means that it keeps going even after it fails and then exits with 0)
 c. Test fails and fail-fast == true (which means it quits immediately with 0 exit code)

I'm not sure you have to do (b), but (c) is pretty important I'd say.